File: src/test/java/com/raychenon/leetcode/graph/CountUnreachablePairsofNodesinanUndirectedGraphTest.java
package com.raychenon.leetcode.graph;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * https://leetcode.com/problems/count-unreachable-pairs-of-nodes-in-an-undirected-graph/
 */
public class CountUnreachablePairsofNodesinanUndirectedGraphTest {

    private static CountUnreachablePairsofNodesinanUndirectedGraph instance;

    @BeforeClass
    public static void setUp() {
        instance = new CountUnreachablePairsofNodesinanUndirectedGraph();
    }

    @Test
    public void example1Test() {
        int n = 3;
        int[][] edges = {
                {0, 1},
                {0, 2},
                {1, 2}
        };
        assertEquals(0, instance.countPairs(n, edges));
    }


    @Test
    public void example2Test() {
        int n = 7;
        int[][] edges = {
                {0, 2},
                {0, 5},
                {2, 4},
                {1, 6},
                {5, 4}
        };
        assertEquals(14, instance.countPairs(n, edges));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/graph/CountUnreachablePairsofNodesinanUndirectedGraphTest.kt
package com.raychenon.kotlin.leetcode.graph

import junit.framework.TestCase
import org.junit.Test

class CountUnreachablePairsofNodesinanUndirectedGraphTest {

    @Test
    fun example1Test() {
        val n = 3
        val edges = arrayOf(intArrayOf(0, 1), intArrayOf(0, 2), intArrayOf(1, 2))
        TestCase.assertEquals(0, CountUnreachablePairsofNodesinanUndirectedGraph.countPairs(n, edges))
    }


    @Test
    fun example2Test() {
        val n = 7
        val edges = arrayOf(intArrayOf(0, 2), intArrayOf(0, 5), intArrayOf(2, 4), intArrayOf(1, 6), intArrayOf(5, 4))
        TestCase.assertEquals(14, CountUnreachablePairsofNodesinanUndirectedGraph.countPairs(n, edges))
    }

}